cmd/fleet/prepare.go

```diff
@@ -73,17 +73,22 @@ To setup Fleet infrastructure, use one of the available commands.
 					bufio.NewScanner(os.Stdin).Scan()
 				}
 			case fleet.UnknownMigrations:
-				fmt.Printf("################################################################################\n"+
-					"# ERROR:\n"+
-					"#   Your Fleet database has unrecognized migrations. This could happen when\n"+
-					"#   running an older version of Fleet on a newer migrated database.\n"+
-					"#\n"+
-					"#   Unknown migrations: tables=%v, data=%v.\n"+
-					"#\n"+
-					"#   Upgrade Fleet server version.\n"+
-					"################################################################################\n",
-					status.UnknownTable, status.UnknownData)
-				os.Exit(1)
+				// We only perform this check in dev mode because some releases may have
+				// introduced some migrations that were later changed, e.g.:
+				// fleet-v4.4.0 released a migration with a timestamp that was changed in fleet-v4.4.1.
+				if dev {
+					fmt.Printf("################################################################################\n"+
+						"# ERROR:\n"+
+						"#   Your Fleet database has unrecognized migrations. This could happen when\n"+
+						"#   running an older version of Fleet on a newer migrated database.\n"+
+						"#\n"+
+						"#   Unknown migrations: tables=%v, data=%v.\n"+
+						"#\n"+
+						"#   Upgrade Fleet server version.\n"+
+						"################################################################################\n",
+						status.UnknownTable, status.UnknownData)
+					os.Exit(1)
```

Comment:
How about we only exit if `dev` but still show the message as a WARNING instead of ERROR? Just a thought, for troubleshooting weird cases.

Reply:
Thought of that, but what can customers/users do about the warning? can they fix it?

Reply:
Letting us know, so that we are aware. If something is broken, and it's because of this, with the WARNING we would know to check, without it, it'll be a puzzle.

Reply:
True, added the exception for the known unknown and only exiting if in `dev`.